gui/src/graphics/mod.rs

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#[cfg_attr(target_os = "macos", path = "metal.rs")]
+#[cfg_attr(not(target_os = "macos"), path = "vulkan.rs")]
+mod api;
+
+#[cfg(not(target_os = "macos"))]
+pub type DefaultApi = self::api::Vulkan;
+
+#[cfg(target_os = "macos")]
+pub type DefaultApi = self::api::Metal;
+
+pub trait GraphicsApi: Sized + 'static {
+    type PhysicalDevice: PhysicalDevice;
+
+    type InitError: core::error::Error;
+
+    fn init() -> Result<Self, Self::InitError>;
```

Comment:
I think this name cause confusion. When I see this name what I understand is it a module-level initialization like https://docs.rs/log/latest/log/fn.set_logger.html. So better to use standard Rust naming like `new` instead.